### ISSUE TITLE
Multi.insert_all/5 contradicts with Ecto.Repo.insert_all/3 in schema_or_source argument's typespecs

### ISSUE THREAD

**Issue by hamzahejja** [elixir-ecto/ecto#3773]
### Precheck

* Do not use the issues tracker for help or support requests (try Elixir Forum, Stack Overflow, IRC or mailing lists, etc).
* For proposing a new feature, please start a discussion on [elixir-ecto](https://groups.google.com/forum/#!forum/elixir-ecto).
* For bugs, do a quick search and make sure the bug has not yet been reported.
* Finally, be nice and have fun!

### Environment

* Elixir version (elixir -v): Elixir 1.11.4 (compiled with Erlang/OTP 23)
* Database and version (PostgreSQL 9.4, MongoDB 3.2, etc.): MySQL 8.0.26
* Ecto version (mix deps): 3.6.2
* Database adapter and version (mix deps): myxql 0.4.0
* Operating system: MacOS Big Sur

### Current behavior
The following call to `Multi.insert_all/5` with passing a tuple of binary source + module, hence {binary(), module()}, as the **source_or_schema** argument will generate the ElixirLS Dialyzer Warning:  `Function x_fun/3 has no local return.` 
```
Multi.insert_all(:insert_all, {"source", ModuleXYZ}, entries, prefix: prefix)
```

### Expected behavior
Now, looking at the type-specs for the arguments of [Repo.insert_all/3](https://hexdocs.pm/ecto/Ecto.Repo.html#c:insert_all/3), I was able to find:
`schema_or_source :: binary() | {binary(), module()} | module()`, 

Whereas, the type-specs for the arguments of [Multi.insert_all/5](https://hexdocs.pm/ecto/Ecto.Multi.html#insert_all/5) shows that:
`@typep schema_or_source :: binary | {binary | nil, binary} | atom`

The fact that the typespecs for `schema_or_source` is different is confusing, especially since from the documentation of `Multi.insert_all/5`, it says that it **Accepts the same arguments and options as Ecto.Repo.insert_all/3 does.**

**Comment by josevalim** [comment]
It is an error in the typespec. Do you want to send a PR that aligns the typespecs? It should be `binary() | {binary(), module()} | module()` indeed.

**Comment by hamzahejja** [comment]
https://github.com/elixir-ecto/ecto/blob/master/lib/ecto/multi.ex#L126
Wait a sec, `multi.ex` file has the correct version of the typespecs for **schema_or_source** 🤔 . Could it be that it was fixed in a later version than the one I have locally in my project?

**Comment by josevalim** [comment]
Yes, fixed in 3.7.0: https://github.com/elixir-ecto/ecto/commit/cff9609f3091fdf3b9eacaf082f18e126177b6fe
